src/js/media-controller.js

```diff
@@ -592,6 +604,14 @@ class MediaController extends MediaContainer {
         this.#hotKeys.value = newValue;
     } else if (attrName === 'default-stream-type') {
       this.propagateMediaState(MediaUIAttributes.MEDIA_STREAM_TYPE);
+    } else if (attrName === 'fullscreen-element') {
+      const el = newValue 
+        // @ts-ignore
+        ? this.getRootNode()?.getElementById(newValue) 
+        : undefined;
+      // NOTE: Setting the internal private prop here to not
+      // clear the attribute that was just set (CJP).
+      this.#fullscreenElement = el;
```

Comment:
Probably overkill, but I'm wondering if we should exit fullscreen on any existing real element when changing this value.

Reply:
I think it should be fine and I kinda like that this doesn't have any direct visible side effects. I know I'd be annoyed if I had to worry about my programmatic workflow unavoidably having a user experience impact.